test/Microsoft.Sbom.Api.Tests/Utils/SPDXFormatDetectorTests.cs

```diff
@@ -36,19 +39,121 @@ public void SetUp()
+            .Setup(m => m.GetSbomFilePath(DirPathStub, Api.Utils.Constants.SPDX22ManifestInfo))
+            .Returns(spdx22FilePathStub)
+            .Verifiable();
+        mockSbomConfigFactory
+            .Setup(m => m.GetSbomFilePath(DirPathStub, Api.Utils.Constants.SPDX30ManifestInfo))
+            .Returns(spdx30FilePathStub)
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.FileExists(It.IsAny<string>()))
+            .Returns(false);
+        mockFileSystemUtils
+            .Setup(m => m.FileExists(FilePathStub + expectedVersion))
+            .Returns(true)
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.OpenRead(FilePathStub + expectedVersion))
+            .Returns(TestUtils.GenerateStreamFromString(testContent))
+            .Verifiable();
+
+        var result = testSubject.TryGetSbomsWithVersion(DirPathStub, out var detectedSboms);
+        Assert.IsTrue(result);
+        Assert.IsNotNull(detectedSboms);
+        Assert.AreEqual(1, detectedSboms.Count());
+        Assert.IsTrue(detectedSboms.Any(value => value.manifestInfo.ToString().Equals(expectedVersion) && value.sbomFilePath.Equals(FilePathStub + expectedVersion)));
+    }
+
+    [TestMethod]
+    public void TryGetSbomsWithVersion_MultipleResults_Success()
+    {
+        var spdx22FilePathStub = FilePathStub + Spdx22VersionStub;
+        var spdx30FilePathStub = FilePathStub + Spdx30VersionStub;
+        mockSbomConfigFactory
+            .Setup(m => m.GetSbomFilePath(DirPathStub, Api.Utils.Constants.SPDX22ManifestInfo))
+            .Returns(spdx22FilePathStub)
+            .Verifiable();
+        mockSbomConfigFactory
+            .Setup(m => m.GetSbomFilePath(DirPathStub, Api.Utils.Constants.SPDX30ManifestInfo))
+            .Returns(spdx30FilePathStub)
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.FileExists(FilePathStub + Spdx22VersionStub))
+            .Returns(true)
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.FileExists(FilePathStub + Spdx30VersionStub))
+            .Returns(true)
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.OpenRead(FilePathStub + Spdx22VersionStub))
+            .Returns(TestUtils.GenerateStreamFromString(Spdx22ContentStub))
+            .Verifiable();
+        mockFileSystemUtils
+            .Setup(m => m.OpenRead(FilePathStub + Spdx30VersionStub))
+            .Returns(TestUtils.GenerateStreamFromString(Spdx30ContentStub))
+            .Verifiable();
+
+        var result = testSubject.TryGetSbomsWithVersion(DirPathStub, out var detectedSboms);
+        Assert.IsTrue(result);
+        Assert.IsNotNull(detectedSboms);
+        Assert.AreEqual(2, detectedSboms.Count());
+        Assert.IsTrue(detectedSboms.Any(value => value.manifestInfo.ToString().Equals(Spdx22VersionStub) && value.sbomFilePath.Equals(FilePathStub + Spdx22VersionStub)));
```

Comment:
If this fails, the unit test will just say something like "expected true, got false", which isn't great for debugging. I try to lean toward 'AreEqual`, or using the comment field to clarify the test failure. Something like
```
Assert.IsTrue(detectedSboms.Any(...), "SPDX 2.2. files should have been detected");
```